ghc-lib-gen/src/Ghclibgen.hs

```diff
@@ -446,33 +446,37 @@ applyPatchGhcPrim ghcFlavor = do
       writeFile tysPrim .
           replace
               "bcoPrimTyCon = pcPrimTyCon0 bcoPrimTyConName LiftedRep"
-              "bcoPrimTyCon = pcPrimTyCon0 bcoPrimTyConName UnliftedRep" .
+              "\n#if __GLASGOW_HASKELL >= 900\nbcoPrimTyCon = pcPrimTyCon0 bcoPrimTyConName LiftedRep\n#else\nbcoPrimTyCon = pcPrimTyCon0 bcoPrimTyConName UnliftedRep\n#endif\n" .
```

Comment:
maybe throw some `unlines` in there and spread the text over multiple lines to make it easier to read?

Reply:
Or even better as suggested below, put it in a helper function.

Reply:
Issue https://github.com/digital-asset/ghc-lib/issues/275

Reply:
i think we'll want `if MIN_VERSION_ghc_prim(0, 7, 0)` in the meantime, the line you comment indicates an error. should be `__GLASGOW_HASKELL__`. i'll push a fix for that.